src/payment/instrument/supported-payment-instruments.ts

```diff
@@ -0,0 +1,42 @@
+import Instrument from './instrument';
+
+interface SupportedInstruments {
+    [key: string]: Partial<Instrument>;
```

Comment:
Maybe `Pick<Instrument, 'provider' | 'method'>` is more accurate so `provider` and `method` are not marked as optional.